packages/manager/modules/pci/src/projects/project/storages/volume-backup/list/create-volume/controller.js

```diff
@@ -0,0 +1,96 @@
+export default class VolumeBackupListCreateVolumeController {
+  /* @ngInject */
+  constructor($translate, ovhManagerRegionService, VolumeBackupService) {
+    this.$translate = $translate;
+    this.ovhManagerRegionService = ovhManagerRegionService;
+    this.volumeBackupService = VolumeBackupService;
+  }
+
+  $onInit() {
+    this.isCreating = false;
+    this.volumeSizeLimit = {
+      min: 10,
+      max: 4000,
+    };
+    this.createVolumeModel = {
+      name: this.volume.name,
+      size: this.volume.size || 0,
+    };
+  }
+
+  buildSuccessMessage() {
+    const volumeSuccessMessage = this.$translate.instant(
+      'pci_projects_project_storages_volume_backup_list_create_volume_request_success',
+      {
+        volumeName: `<strong>${this.createVolumeModel.name}</strong>`,
+      },
+    );
+    const linkMessage = this.$translate.instant(
+      'pci_projects_project_storages_volume_backup_list_create_volume_request_success_link',
+    );
+
+    const volumeListingPageLink = `<a
+      href="${this.volumeBlockStorageLink}"
+      target="_blank"
+      rel="noopener"
+    >
+      <span>${linkMessage}</span>
+      <span
+        class="oui-icon oui-icon-external-link"
+        aria-hidden="true"
+      ></span>
+    </a>`;
+
+    return {
+      textHtml: `${volumeSuccessMessage} <br/> ${volumeListingPageLink}`,
+    };
+  }
+
+  onVolumeSizeChange(volumeSize) {
+    this.getVolumePriceEstimation({
+      ...this.volume,
+      ...{ size: volumeSize },
```

Comment:
Is there a need to deconstruct an object with a single attribute ?

Reply:
Yep, need it to keep volume object unchanged.
so just create an object based on related volume but with new size value.

Reply:
I was referring to `{ size: volumeSize }`. 
couldn't this object be written like : 
`{
  ...this.volume,
  size: volumeSize,
 ....
 `

Reply:
done